index.js

```diff
@@ -0,0 +1,209 @@
+    'lines-between-class-members': 'off',
+    'max-lines': 'error',
+    'no-console': 'error',
+    'no-debugger': 'error',
+    'no-extra-boolean-cast': 'off',
+    'no-plusplus': 'off',
+    'no-restricted-properties': [
+      'error',
+      { object: 'document', property: 'querySelector' },
+      { object: 'document', property: 'querySelectorAll' },
+    ],
+    'no-undef': 'error',
+    'no-underscore-dangle': 'error',
+    'no-unused-expressions': 'off',
+    'react/static-property-placement': 'off',
+    'unicorn/catch-error-name': 'error',
+    'unicorn/custom-error-definition': 'off',
+    'unicorn/error-message': 'error',
+    'unicorn/escape-case': 'error',
+    'unicorn/explicit-length-check': 'error',
+    'unicorn/new-for-builtins': 'error',
+    'unicorn/no-abusive-eslint-disable': 'error',
+    'unicorn/no-array-instanceof': 'error',
+    'unicorn/no-console-spaces': 'error',
+    'unicorn/no-fn-reference-in-iterator': 'off',
+    'unicorn/no-for-loop': 'error',
+    'unicorn/no-hex-escape': 'error',
+    'unicorn/no-new-buffer': 'error',
+    'unicorn/no-process-exit': 'error',
+    'unicorn/no-unreadable-array-destructuring': 'error',
+    'unicorn/no-unsafe-regex': 'off',
+    'unicorn/no-unused-properties': 'error',
+    'unicorn/no-zero-fractions': 'error',
+    'unicorn/number-literal-case': 'error',
+    'unicorn/prefer-exponentiation-operator': 'error',
+    'unicorn/prefer-includes': 'error',
+    'unicorn/prefer-node-remove': 'error',
+    'unicorn/prefer-query-selector': 'error',
+    'unicorn/prefer-starts-ends-with': 'error',
+    'unicorn/prefer-text-content': 'error',
+    'unicorn/prefer-type-error': 'error',
+    'unicorn/prevent-abbreviations': [
+      'error',
+      {
+        replacements: {
+          ref: false,
+          refs: false,
+          prop: false,
+          props: false,
+          src: false,
+          param: false,
+          params: false,
+          args: false,
+        },
+      },
+    ],
+    'unicorn/throw-new-error': 'error',
+  },
+  overrides: [
+    {
+      files: ['*.vue'],
+      parser: 'vue-eslint-parser',
+      rules: {
+        'prettier/prettier': ['off'],
+      },
+    },
+    {
+      files: ['*.ts'],
+      ...typeScriptSettings,
+    },
+    {
+      files: ['*.tsx'],
+      ...typeScriptSettings,
+      extends: [...typeScriptSettings.extends, ...reactSettings.extends],
+      rules: {
+        ...typeScriptSettings.rules,
+        ...reactSettings.rules,
+        'react/prop-types': 'off',
+        '@typescript-eslint/naming-convention': [
+          'error',
+          {
+            selector: 'default',
+            format: ['camelCase'],
+            leadingUnderscore: 'forbid',
+            trailingUnderscore: 'forbid',
+          },
+          {
+            selector: 'typeLike',
+            format: ['PascalCase'],
+          },
+          {
+            selector: 'variable',
+            // Exception for FunctionComponents
```

Comment:
Note, this requires you to write functional components as: `const App = () => {`
instead of `function App() {`, which is the default in CRA.